File: corgi-spark/src/main/scala/com/dyingbleed/corgi/spark/Application.scala
package com.dyingbleed.corgi.spark

import com.dyingbleed.corgi.spark.annotation.EnableMeasure
import com.dyingbleed.corgi.spark.core.Bootstrap
import com.dyingbleed.corgi.spark.ds.EL
import com.google.inject._

/**
  * Created by 李震 on 2018/3/1.
  */
object Application {

  def main(args: Array[String]): Unit = {
    Bootstrap(args).bootstrap(execute)
  }

  def execute(module: AbstractModule): Unit = {
    val injector = Guice.createInjector(module)
    injector.getInstance(classOf[Application]).run()
  }

}

class Application {

  @Inject
  var el: EL = _

  @EnableMeasure
  def run(): Unit = {
    // 加载数据
    val df = el.loadSourceDF

    // 保存数据
    if (Bootstrap.debug) {
      df.explain()
    } else {
      el.persistSinkDF(df)
    }
  }

}

File: corgi-web/src/main/java/com/dyingbleed/corgi/web/Application.java
package com.dyingbleed.corgi.web;

import com.dyingbleed.corgi.web.configuration.ApplicationConfiguration;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Import;

/**
 * Created by 李震 on 2018/4/12.
 */
@SpringBootApplication
@Import(ApplicationConfiguration.class)
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

}
